plugins/pipelines/openPr.js

```diff
@@ -0,0 +1,89 @@
+'use strict';
+
+const boom = require('boom');
+const joi = require('joi');
+const schema = require('screwdriver-data-schema');
+const helper = require('./helper');
+const pipelineCheckoutUrlSchema = joi.reach(schema.models.pipeline.create, 'checkoutUrl');
+const pipelineRootDirSchema = joi.reach(schema.models.pipeline.create, 'rootDir');
+
+module.exports = () => ({
+    method: 'POST',
+    path: '/pipelines/openPr',
+    config: {
+        description: 'Open pull request for repository',
+        notes: 'Open pull request',
+        tags: ['api', 'pipelines'],
+        auth: {
+            strategies: ['token'],
+            scope: ['user', '!guest']
+        },
+        plugins: {
+            'hapi-swagger': {
+                security: [{ token: [] }]
+            }
+        },
+        handler: (request, reply) => {
+            const { userFactory } = request.server.app;
+            const { username, scmContext } = request.auth.credentials;
+            const { files, title, message } = request.payload;
+            const checkoutUrl = helper.formatCheckoutUrl(request.payload.checkoutUrl);
+            const rootDir = helper.sanitizeRootDir(request.payload.rootDir);
+
+            return userFactory
+                .get({ username, scmContext })
+                .then(user => {
+                    if (!user) {
+                        throw boom.notFound(`User ${username} does not exist`);
+                    }
+
+                    return user
+                        .unsealToken()
+                        .then(token =>
+                            userFactory.scm.parseUrl({
+                                scmContext,
+                                rootDir,
+                                checkoutUrl,
+                                token
+                            })
+                        )
+                        .then(scmUri => user.getPermissions(scmUri))
+                        .then(permissions => {
+                            if (!permissions.push) {
+                                throw boom.forbidden(`User ${username} does not have push access for this repo`);
+                            }
+                        })
+                        .then(token =>
+                            userFactory.scm.openPr({
```

Comment:
do you need to implement something like this in scm-gitlab? https://github.com/screwdriver-cd/scm-gitlab/blob/master/index.js#L748-L761

Reply:
yeah we can do that

Reply:
https://github.com/screwdriver-cd/scm-gitlab/pull/28